assets/js/coveo.js

```diff
@@ -1,6 +1,15 @@
 import * as params from '@params';
 document.addEventListener('DOMContentLoaded', function () {
-    Coveo.SearchEndpoint.configureCloudV2Endpoint("", params.coveokey);
+
+    // call our Netlify function via API to get the coveo search key
+    async function getSearchToken() {
+      const response = await fetch(
+        "netlify/functions/search_token"
+      );
+      return await response.json();
+    }
+
```

Comment:
You'll need to `await` the function itself, calling it in line 12 like that I don't think the value will be populated. You could instead have;
`const searchToken = await getSearchToken()`
and update line 12 with 
`Coveo.SearchEndpoint.configureCloudV2Endpoint("", searchToken);`

Reply:
Could I use the `.then()` clause to handle the promise without using `await` at that level?

Reply:
We'd a chat offline and made some tweaks. Once it's validated in a dev environment then we should be good to go :)